RealmSwift/Tests/ProjectionTests.swift

```diff
@@ -0,0 +1,1792 @@
+    @Projected(\ModernAllTypesObject.setOptObjectId) var setOptObjectId
+    @Projected(\ModernAllTypesObject.setOptUuid) var setOptUuid
+
+    @Projected(\ModernAllTypesObject.mapBool) var mapBool
+    @Projected(\ModernAllTypesObject.mapInt) var mapInt
+    @Projected(\ModernAllTypesObject.mapInt8) var mapInt8
+    @Projected(\ModernAllTypesObject.mapInt16) var mapInt16
+    @Projected(\ModernAllTypesObject.mapInt32) var mapInt32
+    @Projected(\ModernAllTypesObject.mapInt64) var mapInt64
+    @Projected(\ModernAllTypesObject.mapFloat) var mapFloat
+    @Projected(\ModernAllTypesObject.mapDouble) var mapDouble
+    @Projected(\ModernAllTypesObject.mapString) var mapString
+    @Projected(\ModernAllTypesObject.mapBinary) var mapBinary
+    @Projected(\ModernAllTypesObject.mapDate) var mapDate
+    @Projected(\ModernAllTypesObject.mapDecimal) var mapDecimal
+    @Projected(\ModernAllTypesObject.mapObjectId) var mapObjectId
+    @Projected(\ModernAllTypesObject.mapAny) var mapAny
+    @Projected(\ModernAllTypesObject.mapUuid) var mapUuid
+
+    @Projected(\ModernAllTypesObject.mapOptBool) var mapOptBool
+    @Projected(\ModernAllTypesObject.mapOptInt) var mapOptInt
+    @Projected(\ModernAllTypesObject.mapOptInt8) var mapOptInt8
+    @Projected(\ModernAllTypesObject.mapOptInt16) var mapOptInt16
+    @Projected(\ModernAllTypesObject.mapOptInt32) var mapOptInt32
+    @Projected(\ModernAllTypesObject.mapOptInt64) var mapOptInt64
+    @Projected(\ModernAllTypesObject.mapOptFloat) var mapOptFloat
+    @Projected(\ModernAllTypesObject.mapOptDouble) var mapOptDouble
+    @Projected(\ModernAllTypesObject.mapOptString) var mapOptString
+    @Projected(\ModernAllTypesObject.mapOptBinary) var mapOptBinary
+    @Projected(\ModernAllTypesObject.mapOptDate) var mapOptDate
+    @Projected(\ModernAllTypesObject.mapOptDecimal) var mapOptDecimal
+    @Projected(\ModernAllTypesObject.mapOptObjectId) var mapOptObjectId
+    @Projected(\ModernAllTypesObject.mapOptUuid) var mapOptUuid
+
+    @Projected(\ModernAllTypesObject.linkingObjects) var linkingObjects
+}
+
+class AdvancedObject: Object {
+    @Persisted(primaryKey: true) var pk: ObjectId
+    @Persisted var commonArray: RealmSwift.List<Int>
+    @Persisted var objectsArray: RealmSwift.List<SimpleObject>
+    @Persisted var commonSet: MutableSet<Int>
+    @Persisted var objectsSet: MutableSet<SimpleObject>
+}
+
+extension SimpleObject {
+    var stringify: String {
+        "\(int) - \(bool)"
+    }
+}
+
+class AdvancedProjection: Projection<AdvancedObject> {
+    @Projected(\AdvancedObject.commonArray.count) var arrayLen
+    @Projected(\AdvancedObject.commonArray) var renamedArray
+    @Projected(\AdvancedObject.objectsArray.projectTo.stringify) var projectedArray: ProjectedList<String>
+    @Projected(\AdvancedObject.commonSet.first) var firstElement
+    @Projected(\AdvancedObject.objectsSet.projectTo.bool) var projectedSet: ProjectedMutableSet<Bool>
```

Comment:
Is every possible type of advanced projection present in this class?

Reply:
Yes